File: vdp-python/visa/test/mlc/test_location_update.py
from visa.helpers.visa_api_client import VisaAPIClient
import json
import sys
import os
import datetime
import unittest
if sys.version_info < (3, 0):
    import ConfigParser as parser
else:
    import configparser as parser
'''
@author: visa
'''

class TestLocationUpdate(unittest.TestCase):
    
    config = parser.ConfigParser()
    config_path = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(__file__)),'..','configuration.ini'))
    config.read(config_path)
    
    def setUp(self):
        date = datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
        self.visa_api_client = VisaAPIClient()
        self.location_update = json.loads('''{
                                    "accuracy": "5000",
                                    "cloudNotificationKey": "03e3ae03-a627-4241-bad6-58f811c18e46",
                                    "cloudNotificationProvider": "1",
                                    "deviceId": "''' + self.config.get('VDP','mlcDeviceId') +'''",
                                    "deviceLocationDateTime": "''' + date + '''",
                                    "geoLocationCoordinate": {
                                        "latitude": "37.558546",
                                        "longitude": "-122.271079"
                                    },
                                    "header": {
                                                "messageDateTime": "''' + date + '''",
                                                "messageId": "''' + self.config.get('VDP','mlcMessageId') +'''"
                                            },
                                    "issuerId": "''' + self.config.get('VDP','mlcIssuerId') +'''",
                                    "provider": "1",
                                    "source": "2"
                                }''')
    
    def test_location_update(self):
        base_uri = 'mlc/'
        resource_path = 'locationupdate/v1/locations' 
        response = self.visa_api_client.do_mutual_auth_request(base_uri + resource_path, self.location_update, 'Location Update Test', 'post')
        self.assertEqual(str(response.status_code) ,"200" ,"Location update test failed")
        pass

File: visa-dev-samplecode-master/vdp-java/src/test/java/com/visa/vdp/mlc/TestLocationUpdate.java
package com.visa.vdp.mlc;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.HashMap;
import java.util.TimeZone;

import org.apache.http.HttpStatus;
import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import com.visa.vdp.utils.VisaAPIClient;
import com.visa.vdp.utils.MethodTypes;
import com.visa.vdp.utils.Property;
import com.visa.vdp.utils.VisaProperties;

public class TestLocationUpdate {

    String locationsRequestBody;
    VisaAPIClient visaAPIClient;

    @BeforeTest(groups = "mlc")
    public void setup() {
        this.visaAPIClient = new VisaAPIClient();
        SimpleDateFormat sdfDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
        TimeZone utc = TimeZone.getTimeZone("UTC");
        sdfDate.setTimeZone(utc);
        Date now = new Date();
        String strDate = sdfDate.format(now);
        this.locationsRequestBody = 
				"{"
					+ "\"accuracy\": \"5000\","
					+ "\"cloudNotificationKey\": \"03e3ae03-a627-4241-bad6-58f811c18e46\","
					+ "\"cloudNotificationProvider\": \"1\","
					+ "\"deviceId\": \""+ VisaProperties.getProperty(Property.MLC_DEVICE_ID)+ "\","
					+ "\"deviceLocationDateTime\": \"" + strDate + "\","
					+ "\"geoLocationCoordinate\": {"
					      + "\"latitude\": \"37.558546\","
					      + "\"longitude\": \"-122.271079\""
					+ "},"
					+ "\"header\": {"
					      + "\"messageDateTime\": \"" + strDate + "\","
					      + "\"messageId\": \""+ VisaProperties.getProperty(Property.MLC_MESSAGE_ID)+ "\""
					+ "},"
					+ "\"issuerId\": \""+ VisaProperties.getProperty(Property.MLC_ISSUER_ID)+ "\","
					+ "\"provider\": \"1\","
					+ "\"source\": \"2\""
				+ "}";
	}

    @Test(groups = "mlc")
    public void testlocations() throws Exception {
        String baseUri = "mlc/";
        String resourcePath = "locationupdate/v1/locations";

        CloseableHttpResponse response = this.visaAPIClient.doMutualAuthRequest(baseUri + resourcePath,"Locations Update Test", this.locationsRequestBody, MethodTypes.POST, new HashMap<String, String>());
        Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
        response.close();
    }

}
